src/third_party/mozjs-60/patches/freebsd-powerpc64le-fix.patch

```diff
@@ -0,0 +1,15 @@
+diff --git a/src/third_party/mozjs-60/extract/js/src/wasm/WasmSignalHandlers.cpp b/src/third_party/mozjs-60/extract/js/src/wasm/WasmSignalHandlers.cpp
+index b35461579c..0cd89e301d 100644
+--- a/src/third_party/mozjs-60/extract/js/src/wasm/WasmSignalHandlers.cpp
++++ b/src/third_party/mozjs-60/extract/js/src/wasm/WasmSignalHandlers.cpp
+@@ -263,6 +263,10 @@ struct AutoSignalHandler
+ #  define EPC_sig(p) ((p)->uc_mcontext.mc_pc)
+ #  define RFP_sig(p) ((p)->uc_mcontext.mc_regs[30])
+ # endif
++# if defined(__FreeBSD__) && defined(__powerpc64__)
++#  define R01_sig(p) ((p)->uc_mcontext.mc_frame[1])
```

Comment:
@pkubaj - Actually, taking a closer look at this, and two small questions:

Upstream has this for  `__FreeBSD__` on `ppc64le`: https://searchfox.org/mozilla-central/rev/40205f1b55f1ba8a26fcbd362c20d843e8718a87/js/src/wasm/WasmSignalHandlers.cpp#212-215

Things that look a little different to me are:
 - They look for different defines for the architecture than you do.
 - Instead of accessing `mc_frame[1]` for `R01` they do `mc_gpr[1]`. This file makes me think they are synonyms though: http://web.mit.edu/freebsd/head/sys/powerpc/include/ucontext.h.

Do you think it would be better to stick a little closer to exactly what upstream has here?

Reply:
From `/usr/include/machine/ucontext.h`:
```
/* GPRs and supervisor-level regs */
#define mc_gpr          mc_frame
#define mc_lr           mc_frame[32]
#define mc_cr           mc_frame[33]
#define mc_xer          mc_frame[34]
#define mc_ctr          mc_frame[35]
#define mc_srr0         mc_frame[36]
#define mc_srr1         mc_frame[37]
#define mc_exc          mc_frame[38]
#define mc_dar          mc_frame[39]
#define mc_dsisr        mc_frame[40]
```

So `mc_frame[1]` and `mc_gpr[1]` are actually exactly the same. If you feel like `mc_gpr[1]` is better, I will obviously change it, but it won't change the behaviour of mozjs.

Reply:
Yeah, I agree they are the same. My question was more about whether you wanted to stick closer to what upstream MozJS has, which is different than yours. But it probably doesn't matter. When we get around to ugprading MozJS, your patch will drop out anyway since it won't be needed. So it probably doesn't matter. I'm OK to merge this with you confirming that you agree the code means the same thing.